scrunch/datasets.py

```diff
@@ -183,23 +183,23 @@ def get_project(project, connection=None):
     connection = _default_connection(connection)
     sub_project = None
 
-    if '|' in project:
-        project_split = project.split('|')
-        project = project_split.pop(0)
-        sub_project = '|' + '|'.join(project_split)
-
     try:
-        ret = connection.projects.by('name')[project].entity
-    except KeyError:
-        try:
-            ret = connection.projects.by('id')[project].entity
-        except KeyError:
-            raise KeyError("Project (name or id: %s) not found." % project)
-
-    _project = Project(ret)
+        if re.match(r'^[0-9a-f]{32}$', project, re.I):
```

Comment:
I think it'd be best if you added a new optional argument to `get_project(id=None)` so that you don't have to be guessing by matching if the thing looks like a UUID. 

If the user has an ID and they know it, they can do `get_project(id="abcdef...")` and it is more explicit on what needs to happen. Also makes it very clear that this is an ID and not a folder name that looks like and ID that could be confused.